1_fetch/src/fetch_wqp_data.R

```diff
@@ -5,22 +5,46 @@
 #' 
 #' @param sitecounts_df data frame containing the site identifiers and total number of
 #' records available for each site. Must contain columns `MonitoringLocationIdentifier`.
-#' @param max_sites_allowed integer indicating the maximum number of sites allowed
-#' in each download group. Defaults to 500.
+#' @param max_sites integer indicating the maximum number of sites allowed in each
+#' download group. Defaults to 500.
+#' @param max_results integer indicating the maximum number of records allowed in
+#' each download group. Defaults to 250,000.
 #' 
 #' @return returns a data frame with columns site id, the total number of records,
 #' (retains the column from `sitecounts_df`), site number, and an additional column 
 #' called `download_grp` which is made up of unique groups that enable use of 
 #' `group_by()` and then `tar_group()` for downloading.
 #' 
-add_download_groups <- function(sitecounts_df, max_sites_allowed = 500) {
+add_download_groups <- function(sitecounts_df, max_sites = 500, max_results = 250000) {
   
+  # Check whether any individual sites have a records count that exceeds `max_results`
```

Comment:
Very good idea!

Reply:
Right now we don't do anything with this information, rather, it's just an FYI. I also don't indicate to the user what they should do with this information in the event that a site does have more records than `max_results`.

Reply:
Excellent point. So in those instances, the single site would be considered its own download group, right? And at this stage, the query should just be one characteristic name. Does it seem likely that a single site for a single characteristic has > 250,000 records? That's an average of 5 records every day since 1900 (`250000/((2022-1900)*365)`) and these are discrete samples. If there is a site with that much data for a single characteristic, I would probably reach out to WQP to make the webservices able to handle the download.

So, I wonder if what we should really add to the message to ensure that people are setting up the queries in a responsible way (in case they change the pipeline and we no longer are assuming one characteristic). A la `If you are not already branching across characteristic names, consider doing so to set up a manageable query to WQP`.

Reply:
`11NPSWRD_WQX-CRLA_LTLMP_ST13` site has 242,270 temperature records, and that's _just_ temperature records. Not sure how many rows you'd get if you were doing other chars too. That's the largest site data I've run into so far and it is a NPS site.

Reply:
Have you been able to download data for that site in one query?

Reply:
> Does it seem likely that a single site for a single characteristic has > 250,000 records?

When I wrote that message originally, I wasn't necessarily thinking of an instance where a single site has >250,000 records (although, wow, `11NPSWRD_WQX-CRLA_LTLMP_ST13`!!). Rather, I was thinking if a user changed the value of `max_results` to something lower, they would run into sites that had more records than `max_results` in which case, yes, those site(s) would then get assigned their own download group. 

I like your idea of including the recommendation about branching in the message!